integration/disk_cleanup_helpers.go

```diff
@@ -151,11 +151,47 @@ func waitUntilDataCleanedUpExtended(
 	return errDataCleanupTimedOut
 }
 
-// nolint: deadcode
-func waitUntilFilesetsCleanedUp(filePathPrefix string, namespace ts.ID, extraShard uint32, waitTimeout time.Duration) error {
+// nolint: deadcode, unused
+func waitUntilNamespacesCleanedUp(testSetup *testSetup, filePathPrefix string, namespace ts.ID, waitTimeout time.Duration) error {
+
+	// The idea would be to have a channel that has notifs (ie, --> func if val 1, 2, or 2)
+	// and pass that to a function that does a function based on that state
+	// this would involve locks on the channel to check the functions aren't called
+	// more than once
+
+	c := make(chan error)
+	go func() {
+		c <- waitUntilNamespacesHaveReset(testSetup)
```

Comment:
Are you wanting to wait for this asynchronously and use the result or synchronously here?